src/credential/credential.ts

```diff
@@ -64,12 +71,202 @@ export interface GoogleOAuthAccessToken {
 }
 
 /**
- * Interface for things that generate access tokens.
+ * Interface that provides Google OAuth2 access tokens used to authenticate
```

Comment:
Pretty much every type and function in this module that doesn't appear in the existing index.d.ts file is an internal API. We need to move them to an `-internal` file at some point too.

Reply:
Yes, I'll do it in a subsequent PR